File: src/com/practice/oldIntel/initialization/Parent.java
package com.practice.oldIntel.initialization;

public class Parent extends GrandParent {
    String age;

    public Parent(String age) {
        this.age = age;
    }

    private static final String TAG = "Parent";

    Parent() {
        System.out.println("Parent.Parent constructor");
    }

    static {
        System.out.println("Parent.static initializer 1");
    }

    static {
        System.out.println("Parent.static initializer 2");
    }

    {
        System.out.println("Parent.instance initializer 1");
    }

    {
        System.out.println("Parent.instance initializer 2");
    }

    {
        System.out.println("Parent.instance initializer 3");
    }
}

File: src/com/practice/oldIntel/kotlin/Parent.kt
package com.practice.oldIntel.kotlin

open class Parent  constructor(i: Int) {

    companion object {
        init {
            println("parent static init 1")
        }

        init {
            println("parent static init 2")
        }

        init {
            println("parent static init 3")
        }
    }

    init {
        println("parent init 1")
    }

    init {
        println("parent init 2")
    }

    init {
        println("parent init 3========$i")
    }

}